kohttp/src/main/kotlin/io/github/rybalkinsd/kohttp/interceptors/RetryInterceptor.kt

```diff
@@ -34,7 +35,13 @@ class RetryInterceptor(
                     delay = performAndReturnDelay(delay)
                 }
                 val response = chain.proceed(request)
-                if (!isRetry(response, attemptsCount)) return response
+
+                val nextTime = calculateNextRetry(response, attemptsCount, delay, chain.readTimeoutMillis())
```

Comment:
I suggest to have two stages here:
1. check if we're going to retry ?
2. calculate next delay.

Probably, we need to modify  `performAndReturnDelay`